### DISCUSSION TITLE
Disabling ReCaptcha with Phone Aunthentication

### DISCUSSION THREAD

**Discussion by Leotheluck** [invertase/react-native-firebase#6290] [Q&A]
Hello everyone,

I have an application which features a phone number login with a code using Firebase Auth, however on iOS devices I am getting a Captcha window interrupting the user flow and that we cannot let in the application :/ I've been struggling for days to find any correct solution and the only thing working I've got to so far is disabling for testing, which I guess is not a correct solution to use.

This is how my Phone auth is currently done:
    
    const signUp = async () => {
            try {
                await firebase.appCheck().activate('ignored', false)
                if (phone === 'test number') {
                    auth().settings.appVerificationDisabledForTesting = true   
                }
                const confirmation = await auth().signInWithPhoneNumber(phone)
                setConfirm(confirmation)
            } [...]
    }

    const confirmCode = async () => {
        try {
            await confirm.confirm(code)
            await UserTracking.logEvent('Confirm Code')
            console.log('confirmed')
            Navigation.replace('Landing')
        }
        [...]
        }

Any help or working tutorial on how to disable ReCaptcha for iOS Phone Auth with React Native would be VERY much appreciated 🙏

Sorry for disturbing and I wish you all a very good day,

Leo

**Comment by mikehardy** [comment]
I believe reCaptcha is automatically put in the flow by firebase-ios-sdk when the cannot be verified as coming from your app 

The verification is in the form of APNS sent to the app in the background.

So, the app is probably not configured for APNS (or cannot receive them, like the Simulator). This is probably your root cause: https://firebase.google.com/docs/auth/ios/phone-auth#enable-app-verification

**Reply by Leotheluck** [reply]
I already configured the app for APNS with the key generated from Apple, but still, either on simulator or on TestFlight with a real phone, the captcha is still appearing 🙁

**Reply by mikehardy** [reply]
You may want to launch the app from Xcode on a real device connected to the computer, and watch either the Xcode console or Console.app for the device to see what the device logs when the APNS comes in (or doesn't!). If the APNS hits the device at all you will see log messages, and they will describe what action the system takes (delivery or non-delivery for some logged reason). Absence of message indicates some mis-configuration, non-delivery may give a clue as to something you can change

**Reply by Leotheluck** [reply]
I did not get any log concerning APNS, what could it be due to ?

**Reply by mikehardy** [reply]
some project configuration, like missing entitlement for background notifications or similar perhaps. There is a fair bit of config to do get an app set up to receive APNS - https://rnfirebase.io/messaging/usage/ios-setup

**Reply by Leotheluck** [reply]
Thank you for your help 🙏, I will get all these steps done and keep you updated on how it renders out

**Reply by Joel-wiz** [reply]
Did this work? We are having the same issue

**Reply by AleshkovDenis** [reply]
@Joel-wiz Did you managed to resolve issue?

**Comment by Vimal1464** [comment]
Any solution please

**Reply by ayush-goyal** [reply]
Have you found a solution for this?

**Reply by AleshkovDenis** [reply]
you need to add correct sha in firebase (you can find it in google play console if your app was uploaded)

**Comment by Vimal1464** [comment]
@ayush-goyal  @Leotheluck 
Recaptcha will be automatically get disable once you added the correct SHA keys from google play console to firebase
it will automatically it will stop the recaptcha verification

**Reply by Xaypanya** [reply]
so what about dev environment sir?

**Reply by Vimal1464** [reply]
@Xaypanya in the dev environment, you cannot disable reCAPTCHA. Solving it is mandatory for testing with a real number in dev mode.
